kayak/twopc_runner.go

```diff
@@ -518,15 +520,42 @@ func (r *TwoPCRunner) verifyLeader(req Request) error {
 }
 
 func (r *TwoPCRunner) decodeLogIndex(data interface{}) (uint64, error) {
-	if index, ok := data.(uint64); ok {
+	// FIXME(xq262144), very very hack, lost precision
+	if data == nil {
+		return 0, ErrInvalidLog
+	}
+
+	b, err := json.Marshal(data)
+	if err != nil {
+		return 0, err
+	}
+
+	var index uint64
+	if err = json.Unmarshal(b, &index); err != nil {
+		return 0, err
+	} else {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
if block ends with a return statement, so drop this else and outdent its block